openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -822,6 +822,20 @@ def delete(self, request, usage_key_str, file_path):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+@method_decorator(non_atomic_requests, name="dispatch")
+@view_auth_classes()
+class LibraryBlockPublishView(APIView):
+    """
+    Commit/publish all of the draft changes made to the component.
+    """
+
+    @convert_exceptions
+    def post(self, request, usage_key_str):
+        key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        api.publish_component_changes(key, request.user)
+        return Response({})
```

Comment:
In the future, we might make this return the updated metadata for the block.

We could also add a "revert single component" API by sending a DELETE request to this same endpoint.

This is good for now though :)